lib/config/config.js

```diff
@@ -309,13 +350,17 @@ exports.getMatryoshka = function (trail) {
  * @param {Object} logObj A mage logger object. Config uses logObj.warn and logObj.info.
  */
 
-exports.initialize = function (logObj) {
+exports.initialize = function () {
 	if (isInitialized) {
 		throw new Error('The config module has already been initialized.');
 	}
 
 	isInitialized = true;
-	logger = logObj;
+
+	// We create a temporary log creator until MAGE provides us with one;
+	// See `setLogger` for more details.
+	const temporaryLogger = loggingService.createLogCreator();
```

Comment:
The point of dependency injection through `initialize` was to make the module more testable. I don't know what would be an elegant alternative to what you've done here though. In any case, injecting the logger through `setLogger` is more elegant as far as I'm concerned than passing it to an `initialize` function, but that's kinda besides the point.

So, not a blocker, but maybe you have some ideas?

Reply:
I don't really see a current purpose to making this injectable... if we had to, I guess we could externalize that part into a function.

Reply:
We can do that if/when it becomes an issue 👍